src/shogun/preprocessor/FisherLDA.cpp

```diff
@@ -250,7 +251,10 @@ bool CFisherLDA::fit(CFeatures *features, CLabels *labels, int32_t num_dimension
 		// x=M
 		// MatrixXd M=Sw.householderQr().solve(Sb);
 		// calculate the eigenvalues and eigenvectors of M.
-		EigenSolver<MatrixXd> es(Sw.householderQr().solve(Sb));
+		SGMatrix<float64_t> M=
+				linalg::qr_solver(SGMatrix<float64_t>(Sw), SGMatrix<float64_t>(Sb));
+		SGMatrix<float64_t>::EigenMatrixXtMap M_eig=M;
+		EigenSolver<MatrixXd> es(M_eig);
 
```

Comment:
@lisitsyn suggested that this might work implicitly, i.e. without the explicit cast to SGMatrix, but with an implicit one.
The casting back to eigen should happen in the same line I think